plugins/inputs/snmp/snmp_test.go

```diff
@@ -640,8 +640,8 @@ func TestGather(t *testing.T) {
 	assert.Len(t, m.Fields, 2)
 	assert.Equal(t, 234, m.Fields["myfield2"])
 	assert.Equal(t, "baz", m.Fields["myfield3"])
-	assert.True(t, tstart.Before(m.Time))
-	assert.True(t, tstop.After(m.Time))
+	assert.True(t, tstart.Before(m.Time) || tstart.Equal(m.Time))
+	assert.True(t, tstop.After(m.Time) || tstop.Equal(m.Time))
```

Comment:
For better readability this should be `! tstart.After(m.Time)` or `m.time.After(tstart)` IMO. Same for `tstop`.

Reply:
Corrected.